test/LintStyle.lean

```diff
@@ -134,4 +134,13 @@ lemma foo' : True := trivial
 
 -- TODO: add terms for the term form
 
+/--
+warning: The declID 'double__underscore' contains '__', which does not follow naming conventions.
+Consider using single underscores instead.
+note: this linter can be disabled with `set_option linter.style.check_declID false`
+-/
+#guard_msgs in
+set_option linter.style.check_declID true in
+theorem double__underscore : True := trivial
```

Comment:
How about we add a check that also verifies that the linter checks the `to_additive` name?  E.g.
* a duplicated namespace arising from additivizing (such as `@[to_additive] theorem add.mul`) and
* a double warning arising from `namespace XX @[to_additive] theorem mul__more`.

Reply:
That makes sense. Will add when it works again.
Also the test here should now catch the add.add, right? https://github.com/leanprover-community/mathlib4/blob/3ae6376f67d90e32a4dfbee23157ac3ac4b5508c/test/Lint.lean#L28